src/system/SystemPacketBuffer.h

```diff
@@ -338,6 +338,17 @@ class DLL_EXPORT PacketBuffer : private pbuf
      */
     CHECK_RETURN_VALUE PacketBufferHandle Last();
 
+    /**
+     * Copies data from the payload of a packet buffer. Supports chained buffers.
+     *
+     * @param[in]  buf             Destination buffer; must be at least @a length bytes.
+     * @param[in]  length          Destination buffer length.
+     *
+     * @retval #CHIP_ERROR_BUFER_TOO_SMALL  If the payload length is less than the requested @a length.
+     * @retval #CHIP_NO_ERROR               If the requested payload has been copied.
+     */
+    CHIP_ERROR Read(uint8_t * buf, size_t length);
```

Comment:
```suggestion
    CHIP_ERROR Read(uint8_t * buf, size_t length);
    template <size_t N>
    inline CHIP_ERROR Read(uint8_t (&buf)[N]) const
    {
        return Read(buf, N);
    }
```
might be nice to have.  And then we can use it for the stack-buffer read in the TCP code.